geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/CommonParams.scala

```diff
@@ -169,38 +166,18 @@ trait ConverterConfigParam {
 trait OptionalConverterConfigParam extends ConverterConfigParam {
   @Parameter(names = Array("-C", "--converter"), description = "GeoMesa converter specification as a config string, file name, or name of an available converter",
     required = false)
-  var config: String = null
+  var config: String = _
 }
 
 trait RequiredConverterConfigParam extends ConverterConfigParam {
   @Parameter(names = Array("-C", "--converter"), description = "GeoMesa converter specification as a config string, file name, or name of an available converter",
     required = true)
-  var config: String = null
+  var config: String = _
 }
 
 trait OptionalIndexParam extends TypeNameParam {
   @Parameter(names = Array("--index"), description = "Specify a particular index to query", required = false)
-  var index: String = null
-
-  @throws[ParameterException]
-  def loadIndex(ds: GeoMesaDataStore[_, _, _], mode: IndexMode): Option[GeoMesaFeatureIndex[_, _, _]] = {
-    Option(index).filter(_.length > 0).map { name =>
-      val untypedIndices = ds.manager.indices(ds.getSchema(featureName), mode)
-      val indices =
-        untypedIndices.asInstanceOf[Seq[GeoMesaFeatureIndex[_ <: GeoMesaDataStore[_, _, _], _ <: WrappedFeature, _]]]
-      val matched = if (name.indexOf(':') != -1) {
-        // full identifier with version
-        indices.find(_.identifier.equalsIgnoreCase(name))
-      } else {
-        // just index name
-        indices.find(_.name.equalsIgnoreCase(name))
-      }
-      matched.getOrElse {
-        throw new ParameterException(s"Specified index ' $index' not found. " +
-        s"Available indices are: ${indices.map(_.identifier).mkString(", ")}")
-      }
-    }
-  }
+  var index: String = _
```

Comment:
i'm about to do testing...but wondering what happens now when you pass in an index that doesn't exist?

Reply:
so it doesn't throw an exception if the index does not exist...perhaps it should do that?

Reply:
yeah, in the log you get:
```
2018-01-17 22:59:47,421 ERROR [org.locationtech.geomesa.accumulo.index.AccumuloQueryPlanner] Ignoring invalid strategy name: foo. Valid values are z3, z3:5, z2, z2:4, records, records:2
```
should we just throw an exception there? that would affect other queries too, but arguably that would be a good thing

Reply:
yeah i was thinking throw an exception since it's an input error.

Reply:
changed it to throw an exception

Reply:
yay!